test/behavior/bitcast.zig

```diff
@@ -11,21 +12,119 @@ test "@bitCast i32 -> u32" {
 }
 
 fn testBitCast_i32_u32() !void {
-    try expect(conv(-1) == maxInt(u32));
-    try expect(conv2(maxInt(u32)) == -1);
+    try expect(conv_i32(-1) == maxInt(u32));
+    try expect(conv_u32(maxInt(u32)) == -1);
+    try expect(conv_u32(0x8000_0000) == minInt(i32));
+    try expect(conv_i32(minInt(i32)) == 0x8000_0000);
 }
 
-fn conv(x: i32) u32 {
+fn conv_i32(x: i32) u32 {
     return @bitCast(u32, x);
 }
-fn conv2(x: u32) i32 {
+fn conv_u32(x: u32) i32 {
     return @bitCast(i32, x);
 }
 
+test "@bitCast i48 -> u48" {
+    try testBitCast_i48_u48();
+    comptime try testBitCast_i48_u48();
+}
+
+fn testBitCast_i48_u48() !void {
+    if (builtin.zig_backend == .stage2_wasm) return error.SkipZigTest;
+    if (builtin.zig_backend == .stage2_c) return error.SkipZigTest;
+    if (builtin.zig_backend == .stage2_x86_64) return error.SkipZigTest;
+
+    try expect(conv_i48(-1) == maxInt(u48));
+    try expect(conv_u48(maxInt(u48)) == -1);
+    try expect(conv_u48(0x8000_0000_0000) == minInt(i48));
+    try expect(conv_i48(minInt(i48)) == 0x8000_0000_0000);
+}
+
+fn conv_i48(x: i48) u48 {
+    return @bitCast(u48, x);
+}
+
+fn conv_u48(x: u48) i48 {
+    return @bitCast(i48, x);
+}
+
+test "@bitCast i27 -> u27" {
+    try testBitCast_i27_u27();
+    comptime try testBitCast_i27_u27();
+}
+
+fn testBitCast_i27_u27() !void {
+    if (builtin.zig_backend == .stage2_wasm) return error.SkipZigTest;
+    if (builtin.zig_backend == .stage2_c) return error.SkipZigTest;
+    if (builtin.zig_backend == .stage2_x86_64) return error.SkipZigTest;
+
+    try expect(conv_i27(-1) == maxInt(u27));
+    try expect(conv_u27(maxInt(u27)) == -1);
+    try expect(conv_u27(0x400_0000) == minInt(i27));
+    try expect(conv_i27(minInt(i27)) == 0x400_0000);
+}
+
+fn conv_i27(x: i27) u27 {
+    return @bitCast(u27, x);
+}
+
+fn conv_u27(x: u27) i27 {
+    return @bitCast(i27, x);
+}
+
+test "@bitCast i512 -> u512" {
+    try testBitCast_i512_u512();
+    comptime try testBitCast_i512_u512();
+}
+
+fn testBitCast_i512_u512() !void {
```

Comment:
`2^{9} = 512`, so 512 is no exotic integer. Is this intended, because there were no tests for regular integers?

Reply:
This is intended, because there were no tests for integers larger than a Limb (`usize`)